File: daily/Arrays/922.Sort-array-by-parity-II.py
from typing import List


class Solution:
    def sortArrayByParityII(self, nums: List[int]) -> List[int]:
        n = len(nums)
        even, odd = 0, 1
        while even < n and odd < n:
            if nums[even] % 2 == 0:
                even += 2
            elif nums[odd] % 2 != 0:
                odd += 2
            else:
                nums[even], nums[odd] = nums[odd], nums[even]
                even += 2
                odd += 2
        return nums

File: daily/Arrays/922.Sort-array-by-parity-II.go
package daily

func sortArrayByParityII(nums []int) []int {
	even, odd := 0, 1
	n := len(nums)
	for even < n && odd < n {
		if nums[even]%2 == 0 {
			even += 2
		} else if nums[odd]%2 != 0 {
			odd += 2
		} else {
			nums[even], nums[odd] = nums[odd], nums[even]
			even += 2
			odd += 2
		}
	}
	return nums
}
